### PR TITLE
fix incorrect rename file error message

### PR DESCRIPTION
**Pull request by Manish-210**
### Version
- [ ] Ensure to run `make getrev` to update gosdk version info.

### Changes
-renameworker.go to handle errors arriving from go routine

### Fixes
-https://github.com/0chain/zboxcli/issues/240
(renaming file error)

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | ABILITY TO SKIP TESTS IS DISABLED. SKIP_TESTS OPTION IGNORED.  |
| system-tests  | incorrect_rename_file_error_message  |
| gosdk  | fix_incorrect_rename_file_error_message  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | incorrect_rename_file_error_message  |
| gosdk  | fix_incorrect_rename_file_error_message  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | incorrect_rename_file_error_message  |
| gosdk  | fix_incorrect_rename_file_error_message  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by guruhubb — APPROVED** [review]

File: zboxcore/sdk/renameworker.go [hunk 1]
```diff
@@ -161,10 +165,27 @@ func (req *RenameRequest) ProcessRename() error {
...
+				case wgErrors <- err:
+				default:
+			}
 			l.Logger.Error(err.Error())
 		}(i)
 	}
-	req.wg.Wait()
+
+	go func() {
+		req.wg.Wait()
```
Comment (dabasov) [inline zboxcore/sdk/renameworker.go:177]: it can cause goroutines leakage, if there are a lot of requests with errors, this routine will run forever

**Pull request merged**